app/Traits/SyncsMedia.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Traits;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Collection;
+
+trait SyncsMedia
+{
+    protected function syncMedia(Model $model, Collection $upstream, String $field_name = 'media'): void
```

Comment:
No mention of `$this` in this method -- should probably be static.

Alternatively, it could be applied to individual Models (but I think the current approach is fine)

```suggestion
    public static function syncMedia(Model $model, Collection $upstream, String $field_name = 'media'): void
```